### ISSUE TITLE
tuple_array_conversions simple conflict with rust-lang

### ISSUE THREAD

**Issue by jalexstark** [rust-lang/rust-clippy#15973]
### Summary

When following the instructions in https://doc.rust-lang.org/std/primitive.array.html, clippy issue is raised.

Quotes:

>  There are two syntactic forms for creating an array:
>
 > * A list with each element, i.e., `[x, y, z]`.

and 

> `let bytes: [u8; 3] = [1, 0, 2];`

### Lint Name

tuple_array_conversions

### Reproducer

I tried this code:

```rust
let x: [f64; 4] = [0.0, 1.0, 2.0, 3.0];
let y: [f64; 4] = [9.0, 8.0, 7.0, 6.0];

let pts: [[f64; 4]; 2] = [x, y];
```

(With caveat that I constructed equivalent code. The line `let pts: [[f64; 4]; 2] = [x, y];` and output are copy-and-paste.)

I saw this happen:

```
warning: it looks like you're trying to convert a tuple to an array
    --> tests/spartan_svg_test.rs:3314:38
     |
3314 |             let pts: [[f64; 4]; 2] = [x, y];
     |                                      ^^^^^^
     |
     = help: use `.into()` instead, or `<[T; N]>::from` if type annotations are needed
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#tuple_array_conversions
```

I expected to see this happen:

Beautiful silence.

### Version

```text
rustc 1.90.0 (1159e78c4 2025-09-14)
binary: rustc
commit-hash: 1159e78c4747b02ef996e55082b704c09b970588
commit-date: 2025-09-14
host: x86_64-unknown-linux-gnu
release: 1.90.0
LLVM version: 20.1.8
```

**Comment by profetia** [comment]
I cannot reproduce this issue. See in the [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=af3e21b96eef8d8a618e5444e862aaea)
